src/MSALAuthenticationProvider.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module MSALAuthenticationProvider
+ */
+
+import { AuthenticationProvider } from "./IAuthenticationProvider";
+import { UserAgentApplication } from "msal";
+
+/**
+ * @class
+ * Class representing MSALAuthenticationProvider
+ * @extends AuthenticationProvider
+ */
+export class MSALAuthenticationProvider implements AuthenticationProvider {
+
+    /**
+     * @private
+     * A member holding the clientId of an application
+     */
+    private clientId: string;
+
+    /**
+     * @private
+     * A member holding the list of graph scopes
+     */
+    private scopes: string[];
+
+    /**
+     * @private
+     * A member holding an instance of UserAgentApplication returned from MSAL
+     */
+    private userAgentApplication: UserAgentApplication;
+
+    /**
+     * @constructor
+     * Creates an instance of MSALAuthenticationProvider
+     * @param {string} clientId - The clientId value of an application
+     * @param {string[]} scopes - An array of graph scopes
+     * @param {any} [options] - An options object for MSAL initialization
+     * @returns An instance of MSALAuthenticationProvider
+     */
+    constructor(clientId: string, scopes: string[], options?: any) {
+        const callback = (errorDesc, token, error, tokenType) => {
```

Comment:
no default implementation of this callback?

Reply:
There is no default implementation.